moz-webgpu-cts/src/main.rs

```diff
@@ -398,18 +398,30 @@ fn run(cli: Cli) -> ExitCode {
                         .map(BufReader::new)
                         .map_err(Report::msg)
                         .wrap_err("failed to open file")
-                        .and_then(|reader| {
-                            serde_json::from_reader::<_, ExecutionReport>(reader)
-                                .map(Some)
-                                .or_else(|e| {
-                                    if e.is_eof() && matches!((e.line(), e.column()), (1, 0)) {
-                                        Ok(None)
-                                    } else {
-                                        Err(e)
-                                    }
-                                })
+                        .and_then(|mut reader| {
+                            // Servo has multiple reports in same json file (each in one line)
+                            // First one is from first run, second one run that runs only unexpected results for filtering
```

Comment:
> suggestion(if-minor): If this behavior is Servo-specific, I think it might be better to gate it behind a `matches!(browser, Browser::Servo)` check?

It is, but it should be compatible with normal wptreports files (from firefox). Actually servo's wptreport.json can also contain only one object (one run, no unexpected results, or no filtering requested).

Reply:
@sagudev: I understand that this is backwards-compatible with existing `wptreport.json` documents, and I agree that it is a nice property. However, I think that silently behavior should still be opt-in on some basis. If there were multiple concatenated JSON values in a vanilla `wptreport.json`, that would more likely mean that something has gone wrong (though I suppose the structure of each value would be checked 🤔). I'd rather an error or warning be emitted by default, for now.

Reply:
> It is, but it should be compatible with normal wptreports files (from firefox). Actually servo's wptreport.json can also contain only one object (one run, no unexpected results, or no filtering requested).

I guess I'm confused about this use case's justification, then. This feature seems like it'd only make sense if you could _only_ provide a JSONL document to `moz-webgpu-cts`, but you're noting that Servo emits a single JSON object per document, too. Is there a reason that you want to feed concatenated JSON objects in a single file to `moz-webgpu-cts`, instead of providing it multiple file paths?

Reply:
> RE: justification: I'm guessing that there's an intermediate step in Servo's process that divides a single report into two partitioned JSON objects?

Yes, and given that in the future we can also use those results for intermittent, this additional step in servo would only make it harder.

CI runs do filtering, hence you get to objects in same JSON file, but local runs are usually done without filtering (less CLI flags needed) so it only creates one object in JSON file.

Reply:
I've added warning if there are any leftover objects in JSON (if not servo) in https://github.com/ErichDonGubler/moz-webgpu-cts/commit/905b6b845aca868ca32a4e01561646b032a242cc